app/storage/storage_medium.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from abc import ABCMeta, abstractmethod
+
+logger = logging.getLogger(__name__)
+
+
+class StorageMedium(metaclass=ABCMeta):
```

Comment:
Agree, but as per our offline conversation, think that this would be better off moving to removing this abstract model, making encrypted storage a child of database storage. Given these two classes don't actually share the same method specification.